library/src/blas1/rocblas_scal_strided_batched.hpp

```diff
@@ -0,0 +1,109 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#ifndef __SCAL_STRIDED_BATCHED_HPP__
+#define __SCAL_STRIDED_BATCHED_HPP__
+
+#include "handle.h"
+#include "logging.h"
+#include "rocblas.h"
+#include "scal_host.hpp"
+#include "utility.h"
+
+template <typename T, typename = T>
+constexpr char rocblas_scal_name[] = "unknown";
+template <>
+constexpr char rocblas_scal_name<float>[] = "rocblas_sscal_strided_batched";
+template <>
+constexpr char rocblas_scal_name<double>[] = "rocblas_dscal_strided_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_float_complex>[] = "rocblas_cscal_strided_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_double_complex>[] = "rocblas_zscal_strided_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_float_complex, float>[]
+    = "rocblas_csscal_strided_batched";
+template <>
+constexpr char rocblas_scal_name<rocblas_double_complex, double>[]
+    = "rocblas_zdscal_strided_batched";
+
```

Comment:
`static` needed in header files.

Reply:
If `rocblas_scal_strided_batched.hpp` is only used in `rocblas_scal_strided_batched.cpp`, then it should be hand-inlined into `rocblas_scal_strided_batched.cpp`, and `rocblas_scal_strided_batched.hpp` should be removed. Same for non-strided-batched.

Those changes were recently made wholesale across BLAS in the last few months, so this BLAS1 change would be undoing those changes.

We also don't want to define variables in headers, and we do not want a header file to `#include` all of the other headers needed by the corresponding `.cpp` file -- the `.cpp` and `.hpp` files should each only `#include` what they need, and one of them should not rely on another to `#include` files it needs. We have a script which *enforces* this at the header-file level now.

We also want to use templates as much as possible to reduce duplicate code. `batched` and `strided_batched` and non-batched should all be able to be templatized into a common code,.